nixos/modules/programs/ssh.nix

```diff
@@ -26,10 +26,103 @@ let
+          This is useful for running graphical programs on the remote machine and have them display to your local X11 server.
+          Historically, this value has depended on the value used by the local sshd daemon, but there really isn't a relation between the two.
+          Note: there are some security risks to forwarding an X11 connection.
+          NixOS's X server is built with the SECURITY extension, which prevents some obvious attacks.
+          To enable or disable forwarding on a per-connection basis, see the -X and -x options to ssh.
+          The -Y option to ssh enables trusted forwarding, which bypasses the SECURITY extension.
+        '';
+      };
+    };
+  };
+
+  commaSeparatedSettings = [
+    "CASignatureAlgorithms"
+    "Ciphers"
+    "HostKeyAlgorithms"
+    "HostbasedAcceptedAlgorithms"
+    "KexAlgorithms"
+    "MACs"
+    "PreferredAuthentications"
+    "ProxyJump"
+    "PubkeyAcceptedKeyTypes"
+  ];
+  spaceSeparatedSettings = [
+    "DynamicForward"
+    "GlobalKnownHostsFile"
+    "IPQoS"
+    "LocalForward"
+    "RemoteForward"
+    "UserKnownHostsFile"
+    "CanonicalDomains"
+    "CanonicalizePermittedCNAMEs"
+    "SendEnv"
+    "SetEnv"
+    "IdentityFile"
+    "Include"
+    "LogVerbose"
+    "PermitRemoteOpen"
+    "RevokedHostKeys"
+  ];
+
+  # Renders a single value for openssh `ssh_config` with key and value and indented by 2 spaces.
+  renderSingleValue = settingName: settingValue: "  ${settingName} " +
+    (if lib.isInt settingValue then toString settingValue
+    else if lib.isString settingValue then settingValue
+    else if lib.isBool settingValue && settingValue then "yes"
+    else if lib.isBool settingValue && !settingValue then "no"
+    else if lib.isList settingValue && builtins.elem settingName commaSeparatedSettings then builtins.concatStringsSep "," settingValue
+    else if lib.isList settingValue && builtins.elem settingName spaceSeparatedSettings then builtins.concatStringsSep " " settingValue
+    else if lib.isList settingValue then throw "list value for unknown key ${settingName}: ${(lib.generators.toPretty {}) settingValue}"
+    else throw "unsupported value for key `${settingName}`: ${(lib.generators.toPretty {}) settingValue}")
+  ;
+
+  # Renders a single block for `ssh_config`
+  buildSettingsBlock = blockType: blockName: blockSettings: "${blockType} ${blockName}\n" +
+    lib.concatStringsSep "\n"
+      (lib.mapAttrsToList renderSingleValue (lib.filterAttrs (n: v: n != "_module" && v != null) blockSettings));
+  orderableBlocks = blockName: blockSettings: {
+    inherit blockName;
+    inherit (blockSettings) priority;
+    blockSettings = builtins.removeAttrs blockSettings [ "priority" ];
+  };
+  renderSettingsBlocks = blockType: settingsBlock: lib.concatStringsSep "\n\n" (map (block: buildSettingsBlock blockType block.blockName block.blockSettings) (lib.sortProperties (lib.mapAttrsToList orderableBlocks settingsBlock)));
```

Comment:
This is a little difficult to read however, please split that across multiple lines.